crates/astria-sequencer/src/transaction/checks.rs

```diff
@@ -57,17 +57,15 @@ pub(crate) async fn check_balance_mempool<S: StateReadExt + 'static>(
     state: &S,
 ) -> anyhow::Result<()> {
     let signer_address = crate::address::base_prefixed(tx.verification_key().address_bytes());
-    check_balance_for_total_fees(tx.unsigned_transaction(), signer_address, state).await?;
+    check_balance_for_total_fees_and_transfers(tx.unsigned_transaction(), signer_address, state)
+        .await?;
     Ok(())
 }
 
-// Checks that the account has enough balance to cover the total fees and transferred values
-// for all actions in the transaction.
-pub(crate) async fn check_balance_for_total_fees<S: StateReadExt + 'static>(
+pub(crate) async fn get_fees_for_transaction<S: StateReadExt + 'static>(
     tx: &UnsignedTransaction,
-    from: Address,
     state: &S,
-) -> anyhow::Result<()> {
+) -> anyhow::Result<HashMap<asset::IbcPrefixed, u128>> {
```

Comment:
Fine for doing this in a follow-up: cant we request all the data bits in this method concurrently instead of one after the other?

Reply:
yes - can make an issue